services/user-management/src/main/java/com/foodie/user/controllers/AuthenticationController.java

```diff
@@ -1,74 +1,74 @@
-package com.foodie.user;
+package com.foodie.user.controllers;
 
+import com.foodie.enums.ResponseKeys;
 import org.springframework.web.bind.annotation.RestController;
 
-import com.foodie.jwt.AuthenticationService;
-import com.foodie.models.contracts.AuthenticationRequest;
-import com.foodie.models.contracts.AuthenticationResponse;
-import com.foodie.models.contracts.RegisterRequest;
-import com.foodie.models.contracts.ResponseHandler;
-import com.foodie.models.users.User;
+import com.foodie.user.jwt.AuthenticationService;
+import com.foodie.user.model.User;
+import com.foodie.user.model.contracts.AuthenticationRequest;
+import com.foodie.user.model.contracts.AuthenticationResponse;
+import com.foodie.user.model.contracts.RegisterRequest;
 
 import jakarta.servlet.http.HttpServletRequest;
 import jakarta.servlet.http.HttpServletResponse;
 import jakarta.validation.Valid;
 import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
 
 import org.springframework.web.bind.annotation.RequestMapping;
 
 import java.io.IOException;
+import java.util.Collections;
 
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.RequestBody;
 
+@Slf4j
 @RestController
-@RequestMapping("/api/v1/auth")
 @RequiredArgsConstructor
+@RequestMapping("/api/v1/auth")
 public class AuthenticationController {
 
-    @Autowired
     private final AuthenticationService service;
 
-    private final Logger logger = LoggerFactory.getLogger(RoleController.class);
-
     @PostMapping(path = "/register", produces = "application/json", consumes = "application/json")
     public ResponseEntity<?> registerUser(@Valid @RequestBody RegisterRequest request) {
 
-        logger.info("AuthenticationController.registerUser() called with request: " + request);
+        log.info("AuthenticationController.registerUser() called with request: " + request);
 
         try {
             User registedUser = service.register(request);
 
-            return ResponseHandler.generateResponse(null, registedUser, HttpStatus.CREATED);
+            return ResponseEntity.status(HttpStatus.CREATED)
+                    .body(Collections.singletonMap(ResponseKeys.data.toString(), registedUser));
         } catch (Exception e) {
```

Comment:
It's bad Practice to catch `Èxception` class

Reply:
Ok. What should I catch here

Reply:
Ideally, you'll have a custom-exception class which will extends maybe RuntimeException class and then catch that custom class. Or just straighout catch RuntimeException here

Reply:
Noted